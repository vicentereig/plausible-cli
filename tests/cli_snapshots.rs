use assert_cmd::prelude::*;
use insta::assert_snapshot;
use std::process::Command;
use tempfile::TempDir;

fn command(temp: &TempDir) -> Command {
    let mut cmd = Command::new(assert_cmd::cargo::cargo_bin!("plausible"));
    cmd.env("HOME", temp.path())
        .env("PLAUSIBLE_CLI_DISABLE_KEYRING", "1")
        .env_remove("XDG_CONFIG_HOME")
        .env_remove("APPDATA")
        .env_remove("LOCALAPPDATA");
    cmd
}

fn run_stdout(temp: &TempDir, args: &[&str]) -> String {
    let assert = command(temp).args(args).assert().success();
    String::from_utf8(assert.get_output().stdout.clone()).expect("utf8 stdout")
}

#[test]
fn accounts_list_empty_snapshot() {
    let temp = TempDir::new().expect("temp dir");
    let output = run_stdout(&temp, &["accounts", "list"]);
    assert_snapshot!("accounts_list_empty", output);
}

#[test]
fn accounts_list_json_after_add() {
    let temp = TempDir::new().expect("temp dir");
    command(&temp)
        .args([
            "accounts",
            "add",
            "--alias",
            "primary",
            "--api-key",
            "key-primary",
            "--label",
            "Primary",
        ])
        .assert()
        .success();

    command(&temp)
        .args([
            "accounts",
            "add",
            "--alias",
            "backup",
            "--api-key",
            "key-backup",
        ])
        .assert()
        .success();

    command(&temp)
        .args(["accounts", "budget", "--alias", "backup", "--daily", "321"])
        .assert()
        .success();

    let output = run_stdout(&temp, &["accounts", "list", "--output", "json"]);
    assert_snapshot!("accounts_list_json", output);
}

#[test]
fn queue_inspect_empty_json() {
    let temp = TempDir::new().expect("temp dir");
    command(&temp)
        .args(["accounts", "add", "--alias", "acct", "--api-key", "key"])
        .assert()
        .success();
    let output = run_stdout(&temp, &["queue", "inspect", "--output", "json"]);
    assert_snapshot!("queue_inspect_empty_json", output);
}

#[test]
fn events_template_json_snapshot() {
    let temp = TempDir::new().expect("temp dir");
    command(&temp)
        .args(["accounts", "add", "--alias", "acct", "--api-key", "key"])
        .assert()
        .success();
    let output = run_stdout(&temp, &["events", "template", "--output", "json"]);
    assert_snapshot!("events_template_json", output);
}
