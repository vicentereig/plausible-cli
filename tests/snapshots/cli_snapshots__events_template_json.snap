---
source: tests/cli_snapshots.rs
assertion_line: 78
expression: output
---
{
  "device_type": "desktop",
  "domain": "example.com",
  "name": "Signup",
  "referrer": "https://google.com",
  "url": "https://example.com/signup",
  "utm_medium": "email",
  "utm_source": "newsletter"
}
